File: iminling-core/src/main/java/com/iminling/core/config/mybatis/MapWrapperFactory.java
package com.iminling.core.config.mybatis;

import org.apache.ibatis.reflection.MetaObject;
import org.apache.ibatis.reflection.wrapper.ObjectWrapper;
import org.apache.ibatis.reflection.wrapper.ObjectWrapperFactory;

import java.util.Map;

/**
 * https://my.oschina.net/u/2278977/blog/1795969
 * @author yslao@outlook.com
 */
public class MapWrapperFactory implements ObjectWrapperFactory {
    @Override
    public boolean hasWrapperFor(Object object) {
        return object != null && object instanceof Map;
    }

    @Override
    public ObjectWrapper getWrapperFor(MetaObject metaObject, Object object) {
        return new CustomizeWrapper(metaObject,(Map)object);
    }
}

File: iminling-core/src/main/kotlin/com/iminling/core/config/mybatis/MapWrapperFactory.kt
package com.iminling.core.config.mybatis

import org.apache.ibatis.reflection.MetaObject
import org.apache.ibatis.reflection.wrapper.ObjectWrapper
import org.apache.ibatis.reflection.wrapper.ObjectWrapperFactory

/**
 * https://my.oschina.net/u/2278977/blog/1795969
 * @author  yslao@outlook.com
 * @since  2021/11/28
 */
class MapWrapperFactory : ObjectWrapperFactory {

    override fun hasWrapperFor(`object`: Any?): Boolean {
        return `object` != null && `object` is Map<*, *>?
    }

    override fun getWrapperFor(metaObject: MetaObject?, `object`: Any?): ObjectWrapper {
        return CustomizeWrapper(metaObject!!, `object` as Map<String, Any?>?)
    }
}